### PR TITLE
Publish automation

### PR DESCRIPTION
**Pull request by flavioheleno**
This PR adds a [Github Action](https://github.com/tsunematsu21/actions-publish-gh-pages) to automate publishing the book to github-pages, closing #108.

### CODE REVIEW

File: .github/workflows/publish-to-github-pages.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+      - name: Generate HTML release
+        run: ./build_release_html.sh
+
+      - name: Publish generated content to GitHub Pages
+        uses: tsunematsu21/actions-publish-gh-pages@v1.0.2
+        with:
+          dir: BookHTML
+          branch: gh-pages
+          token: ${{ secrets.ACCESS_TOKEN }}
```
Comment (nikic) [inline .github/workflows/publish-to-github-pages.yml:23]: I don't think this is necessary, as we're deploying to the same repo and can use the default token.

Comment (flavioheleno) [reply .github/workflows/publish-to-github-pages.yml]: Indeed, I've obviously overlooked the fact that the target repository was the same as the source one.

I've updated the yaml file accordingly :)

**Comment by nikic** [comment]
Doesn't seem to be working, though I don't understand the error: https://github.com/phpinternalsbook/PHP-Internals-Book/runs/4781019969?check_suite_focus=true

**Comment by nikic** [comment]
Ah, the relevant failure is in an earlier step:
> ./build_release_html.sh: line 7: sphinx-build: command not found

We need to install sphinx.

**Comment by flavioheleno** [comment]
First and foremost, sorry!

I'll take a look at both the shell script and at the action to ensure that:

1. The shell script returns non-zero on failure (allowing us to spot errors like this one);
2. The action installs sphinx prior to building the release.

**Comment by flavioheleno** [comment]
@nikic I had to extract the commands from the shell script and put them straight into the action, so that I could use both Python for Sphinx and then PHP for the redirects. Are you ok with that?
You can see it before the PR here: https://github.com/flavioheleno/PHP-Internals-Book/commit/89c3279aa841cb9ec468ecea1a307121afbd1624

**Comment by nikic** [comment]
@flavioheleno Not sure I get why they need to be separated. If you install both Python and PHP first, shouldn't it work?

**Comment by flavioheleno** [comment]
I had no idea you could install both and have them available to next steps! Reverting back to using the shell script :)

**Pull request merged**